tedana/selection/select_comps.py

```diff
@@ -71,828 +46,298 @@ def selcomps(seldict, comptable, mmix, mask, ref_img, manacc, n_echos, t2s, s0,
-    # larger is worse - spike
-    mmix_kurt_z = (mmix_kurt-mmix_kurt[group0].mean()) / mmix_kurt[group0].std()
-    # smaller is worse - drift
-    mmix_std_z = -1 * ((mmix_std-mmix_std[group0].mean()) / mmix_std[group0].std())
-    mmix_kurt_z_max = np.max([mmix_kurt_z, mmix_std_z], 0)
+    kappa_rate = ((np.max(comptable.loc[good_guess, 'kappa']) -
+                   np.min(comptable.loc[good_guess, 'kappa'])) /
+                  (np.max(comptable.loc[good_guess, 'variance explained']) -
+                   np.min(comptable.loc[good_guess, 'variance explained'])))
+    kappa_ratios = kappa_rate * comptable['variance explained'] / comptable['kappa']
+    varex_lb = stats.scoreatpercentile(
+        comptable.loc[good_guess, 'variance explained'], LOW_PERC)
+    varex_ub = stats.scoreatpercentile(
+        comptable.loc[good_guess, 'variance explained'], HIGH_PERC)
 
     """
-    Step 2: Classifiy midk and ignore using separate SVMs for
-    different variance regimes
-    # To render hyperplane:
-    min_x = np.min(spz2);max_x=np.max(spz2)
-    # plotting separating hyperplane
-        ww = clf_.coef_[0]
-        aa = -ww[0] / ww[1]
-        # make sure the next line is long enough
-        xx = np.linspace(min_x - 2, max_x + 2)
-        yy = aa * xx - (clf_.intercept_[0]) / ww[1]
-        plt.plot(xx, yy, '-')
+    Step 3: Get rid of midk components; i.e., those with higher than
+    max decision score and high variance
     """
-    LGR.debug('Attempting to classify midk components')
-    # Tried getting rid of accepting based on SVM altogether,
-    # now using only rejecting
-    toacc_hi = np.setdiff1d(all_comps[utils.andb([fdist <= np.max(fdist[group0]),
-                                                  comptable['rho'] < F025, Vz > -2]) == 3],
-                            np.union1d(group0, rej))
-    temp = utils.andb([spz < 1, Rz < 0,
-                       mmix_kurt_z_max < 5,
-                       Dz > -1, Tz > -1, Vz < 0,
-                       comptable['kappa'] >= F025,
-                       fdist < 3 * np.percentile(fdist[group0], 98)]) == 8
-    toacc_lo = np.intersect1d(to_clf, all_comps[temp])
-    midk_clf, clf_ = do_svm(fproj_arr_val[:, np.union1d(group0, rej)].T,
-                            [0] * len(group0) + [1] * len(rej),
-                            fproj_arr_val[:, to_clf].T,
-                            svmtype=2)
-    midk = np.setdiff1d(
-        to_clf[utils.andb([
-            midk_clf == 1,
-            (comptable['variance explained'][to_clf] >
-             np.median(comptable['variance explained'][group0]))]) == 2],
-        np.union1d(toacc_hi, toacc_lo))
+    max_good_d_score = EXTEND_FACTOR * len(good_guess) * n_decision_metrics
+    midk = acc[(comptable.loc[acc, 'd_table_score'] > max_good_d_score) &
+               (comptable.loc[acc, 'variance explained'] > EXTEND_FACTOR * varex_ub)]
     comptable.loc[midk, 'classification'] = 'rejected'
     comptable.loc[midk, 'rationale'] += 'midk;'
```

Comment:
Is there a reason `midk` gets a semi-colon and `rejected` does not ?

Reply:
@handwerkerd requested that we eliminate `midk` as a classification [here](https://github.com/ME-ICA/tedana/issues/58#issuecomment-433975211), so here we just note it as the rationale for rejection. Classifications are set, while rationales are appended to (hence `+=` and the semicolon to delineate reasons). I tried to append the rationale every time components are rejected or ignored, so that any times when a component is rejected or ignored for multiple reasons they're all logged.